stages/test/test_gunzip.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/python3
+
+import subprocess
+from unittest import mock
+
+import pytest
+import gzip
+
+from osbuild import testutil
+from osbuild.testutil import has_executable
+
+STAGE_NAME = "org.osbuild.gunzip"
+
+
+# Prepare dataset containing good and bad API call parameters
+@pytest.mark.parametrize("test_data, expected_err", [
+    # Bad API parameters
+    ({}, "'path' is a required property"),
+    ({"path": 123}, "123 is not of type 'string'"),
+    # Good API parameters
+    ({"path": "/tmp/"}, ""),
+    ({"path": "/boot/initramfs-5.14.0-247.el9.x86_64.img"}, ""),
+])
+# This test validates only API calls using correct and incorrect queries
+def test_schema_validation_gunzip(stage_schema, test_data, expected_err):
+    test_input = {
+        "type": STAGE_NAME,
+        "options": {
+        }
+    }
+
+    test_input["options"].update(test_data)
+    res = stage_schema.validate(test_input)
+
+    if expected_err == "":
+        assert res.valid is True, f"err: {[e.as_dict() for e in res.errors]}"
+    else:
+        assert res.valid is False
+        testutil.assert_jsonschema_error_contains(res, expected_err, expected_num_errs=1)
+
+
+# This test creates a compressed dummy file then decompresses and
+# compares the content.
+@pytest.mark.skipif(not has_executable("gunzip"), reason="gunzip")
+def test_gunzip_integration(stage_module):
+    fake_tree = "/fake/output/tree"
```

Comment:
Sorry, I missed in my previous review that the `tmp_path` fixture will be needed here so that there is a tmp env to create the fake data. Here and in the followup test. Something like:
```python
diff --git a/stages/test/test_gunzip.py b/stages/test/test_gunzip.py
index f202a0db..161a7fe9 100644
--- a/stages/test/test_gunzip.py
+++ b/stages/test/test_gunzip.py
@@ -1,5 +1,6 @@
 #!/usr/bin/python3
 
+import os
 import subprocess
 from unittest import mock
 
@@ -42,9 +43,10 @@ def test_schema_validation_gunzip(stage_schema, test_data, expected_err):
 # This test creates a compressed dummy file then decompresses and
 # compares the content.
 @pytest.mark.skipif(not has_executable("gunzip"), reason="gunzip")
-def test_gunzip_integration(stage_module):
-    fake_tree = "/fake/output/tree"
-    fake_archive_path = fake_tree +  "/" + "some-file.gz"
+def test_gunzip_integration(tmp_path, stage_module):
+    fake_tree = tmp_path / "fake/output/tree"
+    fake_tree.mkdir(parents=True)
+    fake_archive_path = fake_tree / "some-file.gz"
     fake_file_content = b"Some random text written to a file"
     with gzip.GzipFile(fake_archive_path, "w") as fp:
         fp.write(fake_file_content)
@@ -70,8 +72,9 @@ def test_gunzip_integration(stage_module):
 
 # This test evaluates final binary call and match between its flags and stage args
 @mock.patch("subprocess.run")
-def test_gunzip_cmdline(mock_run, stage_module):
-    fake_tree = "/fake/output/tree"
+def test_gunzip_cmdline(mock_run, tmp_path, stage_module):
+    fake_tree = tmp_path / "fake/output/tree"
+    fake_tree.mkdir(parents=True)
     fake_file = "fake_file.txt"
     fake_archive = f"{fake_file}.gz"
 
@@ -94,5 +97,5 @@ def test_gunzip_cmdline(mock_run, stage_module):
     expected = [
         "gunzip",
         "--stdout",
-    ] + "/fake/output/tree/fake_file.txt.gz"
+    ] + [os.fspath(fake_tree / fake_archive)]
     mock_run.assert_called_once_with(expected, stdout=mock.ANY, check=True)
```
should do

Reply:
Fixed